goshopify.go

```diff
@@ -343,8 +343,19 @@ func (c *Client) doGetHeaders(req *http.Request, v interface{}) (http.Header, er
 	c.attempts = 0
 	c.logRequest(req)
 
+	// copy request body so it can be re-used
+	var body []byte
+	if req.Body != nil {
+		body, err = ioutil.ReadAll(req.Body)
+		defer req.Body.Close()
+	}
+	if err != nil {
+		return nil, err
+	}
+
 	for {
 		c.attempts++
+		req.Body = ioutil.NopCloser(bytes.NewBuffer(body))
```

Comment:
is this ok to do if body is empty?

Reply:
If body is empty then req.Body will be set to an empty bytes.Buffer which is fine